src/bc_protocol/connection.rs

```diff
@@ -55,7 +55,15 @@ impl BcConnection {
 
         let rx_thread = std::thread::spawn(move || {
             let mut context = BcContext::new();
-            while BcConnection::poll(&mut context, &conn, &mut subs).is_ok() {}
+            while {
+                match BcConnection::poll(&mut context, &conn, &mut subs) {
+                    Ok(_) => true,
+                    Err(err) => {
+                        error!("Deserialization error: {:?}", err);
+                        false
+                    }
+                }
+            } {}
```

Comment:
I see what you're doing here, but the "naked" `while` is weird imho.  How about:
```
    let mut result;
    while { result = foo(); result.is_ok() } {}
    error!("Deserialization error: {:?}", result.unwrap_err);
    result.unwrap_err();
```

Reply:
I like being verbose but alright I'll change it haha

Reply:
Done